### PR TITLE
Add claimant validation when claimant is not veteran

### PR DESCRIPTION
**Pull request by leikkisa**
connects #7595

Require that a claimant is selected if the user selects the option that the claimant is not the veteran.

### CODE REVIEW

File: app/models/claim_review_intake.rb [hunk 1]
```diff
@@ -54,16 +61,17 @@ def create_claimant!
     else
       Claimant.create!(
         participant_id: veteran.participant_id,
-        payee_code: need_payee_code? ? "00" : nil,
+        payee_code: nil,
         review_request: detail
       )
     end
+    update_person!
```
Comment (leikkisa) [inline app/models/claim_review_intake.rb]: On the last PR we moved to Claimant.create! directly, but previously used DecisionReview's create_claimants! which also updates the person.

File: app/models/appeal_intake.rb [hunk 2]
```diff
@@ -8,12 +10,27 @@ def ui_hash(ama_enabled)
...
+        participant_id: claimant_participant_id,
+        payee_code: nil,
+        review_request: detail
+      )
+      update_person!
+      detail.save(context: :intake_review)
+    end
+  rescue ActiveRecord::RecordInvalid => _err
+    # propagate the error from invalid column to the user-visible reason
+    if detail.errors.messages[:veteran_is_not_claimant].include?(ClaimantValidator::CLAIMANT_REQUIRED)
```
Comment (pkarman) [inline app/models/appeal_intake.rb]: the logic here is slightly different than in `ClaimReviewIntake` where the `detail.validate` only gets called if there is a `CLAIMANT_REQUIRED` error. Was that intentional?

File: app/models/appeal_intake.rb [hunk 3]
```diff
@@ -8,12 +10,27 @@ def ui_hash(ama_enabled)
...
+      claimant_error = "blank"
```
Comment (pkarman) [inline app/models/appeal_intake.rb]: we've started to use this string often enough that it feels like it should be a constant somewhere.

File: app/models/decision_review.rb [hunk 4]
```diff
@@ -82,6 +82,8 @@ def claimant_participant_id
   end
 
   def claimant_not_veteran
+    # This is being replaced by veteran_is_not_claimant, but keeping it temporarily
```
Comment (pkarman) [inline app/models/decision_review.rb]: +1

File: spec/feature/intake/higher_level_review_spec.rb [hunk 5]
```diff
@@ -8,6 +8,7 @@
     FeatureToggle.enable!(:intake)
     FeatureToggle.enable!(:intakeAma)
     FeatureToggle.enable!(:intake_legacy_opt_in)
+    FeatureToggle.disable!(:intake_enable_add_issues_page)
```
Comment (pkarman) [inline spec/feature/intake/higher_level_review_spec.rb]: are we adding tests to the old intake flow?

File: spec/models/appeal_intake_spec.rb [hunk 6]
```diff
@@ -125,6 +129,18 @@
           payee_code: nil
         )
       end
+
+      context "and claimant is nil" do
```
Comment (pkarman) [inline spec/models/appeal_intake_spec.rb]: a nit: the "and" here seems to be implied everywhere else `context` is used.

File: spec/feature/intake/higher_level_review_spec.rb (hunk 5 above)
Comment (leikkisa) [reply spec/feature/intake/higher_level_review_spec.rb]: I modified the existing test that had this section covered (the review page validation).

File: spec/models/appeal_intake_spec.rb (hunk 6 above)
Comment (leikkisa) [reply spec/models/appeal_intake_spec.rb]: Will remove!

File: app/models/appeal_intake.rb (hunk 2 above)
Comment (leikkisa) [reply app/models/appeal_intake.rb]: I don't see what you're talking about.

Comment (pkarman) [reply app/models/appeal_intake.rb]: I see now that both models have changed in order to escalate the error messages from child object validation into something the view can use. Seems ok for now.

**Pull request merged**